sound/soc/sof/intel/hda.c

```diff
@@ -527,6 +527,8 @@ static int check_dmic_num(struct snd_sof_dev *sdev)
 		dmic_num = intel_nhlt_get_dmic_geo(sdev->dev, nhlt);
 
 	dev_info(sdev->dev, "DMICs detected in NHLT tables: %d\n", dmic_num);
+	dev_info(sdev->dev,
+		 "If the DMIC number doesn't match to the real HW design, please overwrite it with \"dmic_num\" kernel param.\n");
```

Comment:
I would change this too
```c
dev_info(sdev->dev, "DMICs detected in NHLT tables: %d, can be overidden with \"dmic_num\" kernel param if incorrect\n", dmic_num);
```
and I would move the new larger warning to the code block when we detect both SDW and DMIC
```c
dev_info(sdev->dev, "NHLT reporting both SDW and DMIC endpoints, if incorrect, pls override with .....);
```

Reply:
Update: only show the warning when both SDW mic and PCH DMIC are present.